### PR TITLE
Only show grid lines when selecting grid or during child interaction

### LINKED ISSUE #6645: Only show grid lines when the grid is selected or when child interaction is active

### PR DESCRIPTION
**Pull request by ruggi**
**Problem:**

Grid lines are shown all the time whether a grid or a grid child is selected. The grid sizing pills are also shown all the time.

**Fix:**

Only show the grid lines if:
- the grid itself is selected
- a grid child is selected **and** an interaction is in progress

Only show the grid sizing pills if:
- the grid itself is selected

**Note:** I enabled the grid lines also for absolute resize of grid children because they can be affected by the grid so I thought it would be good to have them, but lmk if you feel otherwise.

https://github.com/user-attachments/assets/eba83673-fe5d-43a2-8dcf-aba7bcc2cca1

Fixes #6645

### CODE REVIEW

File: editor/src/components/canvas/controls/grid-controls.tsx [hunk 1]
```diff
@@ -1144,20 +1164,34 @@ export const GridControlsComponent = ({ targets }: GridControlsProps) => {
...
               : EP.parentPath(grid.identifier.path)
-          const shouldHaveVisibleControls = gridsWithVisibleControls.some((g) => {
-            const visibleControlPath = g.type === 'GRID_CONTAINER' ? g.path : EP.parentPath(g.path)
-            return EP.pathsEqual(gridPath, visibleControlPath)
-          })
+
+          function shouldHaveVisibleControls(): boolean {
+            const isGridChildSelectedWihtoutInteraction =
+              gridItemSelected && !isItemInteractionActive
+            if (isGridChildSelectedWihtoutInteraction) {
```
Comment (seanparsons) [inline editor/src/components/canvas/controls/grid-controls.tsx:1179]: Just spotted a typo!

**Comment by balazsbajorics** [comment]
please hide the track size pills too! 

image

**Review by seanparsons — APPROVED** [review]

File: editor/src/components/canvas/controls/grid-controls.tsx (hunk 1 above)
Comment (ruggi) [reply editor/src/components/canvas/controls/grid-controls.tsx]: fixed

**Comment by ruggi** [comment]
> please hide the track size pills too!

@balazsbajorics fixed ✅

File: editor/src/components/canvas/controls/grid-controls.tsx [hunk 2]
```diff
@@ -2078,3 +2092,36 @@ function useAllGrids(metadata: ElementInstanceMetadataMap) {
...
+        store.editor.canvas.interactionSession.activeControl.type === 'RESIZE_HANDLE'
+      )
+    },
+    'useShouldHaveVisibleControls isItemInteractionActive',
+  )
+
+  return React.useCallback(
+    (grid: GridData) => {
+      const gridItemSelected = selectedViewsRef.current.some((path) =>
+        EP.isDescendantOf(path, grid.identifier.path),
```
Comment (gbalint) [inline editor/src/components/canvas/controls/grid-controls.tsx:2120]: This is only true when grid.identifier is a GridContainerIdentifier. Currently on master this works, but after https://github.com/concrete-utopia/utopia/pull/6636 , when a grid item is selected, this is going to be a GridItemIdentifier.

Comment (ruggi) [reply editor/src/components/canvas/controls/grid-controls.tsx:2120]: I'll wait and update once #6640 is in

Comment (ruggi) [reply editor/src/components/canvas/controls/grid-controls.tsx]: done

**Review by gbalint — APPROVED** [review]

**Pull request merged**